src/work/PokerGame/PokerGame.sol

```diff
@@ -0,0 +1,516 @@
+
+        _grantRole(CURRENT_TURN_ROLE, nextPlayer.addr);
+        nextPlayer.hasCurrentTurnRole = true;
+
+        if (isRoundComplete(gameId)) {
+            // Handle end of betting round logic here
+            // This could involve dealing community cards, starting a new betting round, or ending the game
+        }
+
+        emit TurnChanged(gameId, game.players[game.currentPlayerIndex].addr); // Emit the TurnChanged event
+    }
+
+    // Internal function to check if the betting round is complete
+    function isRoundComplete(uint256 gameId) internal view returns (bool) {
+        Game storage game = games[gameId]; // Get the game from storage
+        uint256 activePlayers = 0; // Count of active players
+        uint256 playersActed = 0; // Count of players who have acted
+        uint256 highestBet = 0; // Highest bet in the current round
+
+        for (uint8 i = 0; i < game.playerCount; i++) { // Loop through all players
+            Player storage player = game.players[i];
+            if (player.isActive) {
+                activePlayers++;
+                if (player.lastAction != PlayerAction.NONE) {
+                    playersActed++;
+                }
+                uint256 playerBet = getPlayerBet(gameId, player.addr);
+                if (playerBet > highestBet) {
+                    highestBet = playerBet;
+                }
+            }
+        }
+
+        return (activePlayers == playersActed) && (highestBet == game.highestBet); // Return true if all active players have acted and the highest bet is matched
+    }
+
+    // Internal function to get the total bet amount of a player
+    function getPlayerBet(uint256 gameId, address playerAddress) internal view returns (uint256) {
+        Game storage game = games[gameId]; // Get the game from storage
+        for (uint8 i = 0; i < game.playerCount; i++) { // Loop through all players
+            if (game.players[i].addr == playerAddress) {
+                return INITIAL_CHIP_COUNT - game.players[i].chipCount; // Return the total bet amount
+            }
+        }
+        revert("Player not found"); // Revert if the player is not found
+    }
+
+    function hasCurrentTurn(uint256 gameId, address player) public view returns (bool) {
+        Game storage game = games[gameId];
+        for (uint8 i = 0; i < game.playerCount; i++) {
+            if (game.players[i].addr == player) {
+                return game.players[i].hasCurrentTurnRole;
+            }
+        }
+        return false;
+    }
+    function getCommunityCards(uint256 gameId) external view returns (Card[5] memory) {
+        Game storage game = games[gameId];
+        return [
+            game.communityCards.flop1,
+            game.communityCards.flop2,
+            game.communityCards.flop3,
+            game.communityCards.turn,
+            game.communityCards.river
+        ];
+    }
+    // Function for the dealer to commit community cards
+    function commitCommunityCards(
```

Comment:
Would also be good to have a state on to see if community cards have already been revealed. So you can revert if true.